conan/tools/meson/toolchain.py

```diff
@@ -164,6 +167,45 @@ def __init__(self, conanfile, backend=None):
         self._resolve_apple_flags()
         self._resolve_android_cross_compilation()
 
+    def _get_default_dirs(self):
+        """
+        Get all the default directories from cpp.package.
+
+        Issues related:
+            - https://github.com/conan-io/conan/issues/9713
+            - https://github.com/conan-io/conan/issues/11596
+        """
+        def _get_cpp_info_value(name):
+            elements = getattr(self._conanfile.cpp.package, name)
+            return elements[0] if elements else None
+
+        if not self._conanfile.package_folder:
+            return {}
+
+        ret = {}
+        bindir = _get_cpp_info_value("bindirs")
+        datadir = _get_cpp_info_value("resdirs")
+        libdir = _get_cpp_info_value("libdirs")
+        includedir = _get_cpp_info_value("includedirs")
+        if bindir:
+            ret.update({
+                'bindir': bindir,
+                'sbindir': bindir,
+                'libexecdir': bindir
+            })
+        if datadir:
+            ret.update({
+                'datadir': datadir,
+                'localedir': datadir,
+                'mandir': datadir,
+                'infodir': datadir
+            })
+        if includedir:
+            ret["includedir"] = includedir
+        if libdir:
+            ret["libdir"] = libdir
```

Comment:
at least one test (`conans\test\functional\toolchains\meson\test_install.py`) uses hard-coded `tc.project_options["libdir"] = "lib"`. it would be nice to check this test works without hard-coded value now.

Reply:
yes, sure. I put it as a draft like I did not add any tests yet